### PR TITLE
SQL: Implement CURRENT_TIME/CURTIME functions

### LINKED ISSUE #40648: SQL: Implement CURTIME()/CURRENT_TIME()
Now that SQL TIME type is available implement CURRENT_TIME similar to CURRENT_TIMESTAMP and add its alias CURTIME .

Part of #33092

### PR DESCRIPTION
**Pull request by matriv**
After `TIME` SQL data type is introduced, implement
`CURRENT_TIME/CURTIME` functions similar to CURRENT_TIMESTAMP
that return the system's current time (only, without the date part).

Closes: #40648

### CODE REVIEW

**Comment by matriv** [comment]
Needs to wait for #39802 and be rebased on top.

File: docs/reference/sql/language/data-types.asciidoc [hunk 1]
```diff
@@ -44,9 +44,10 @@ s|SQL precision
 Most of {es} <<mapping-types, data types>> are available in {es-sql}, as indicated above.
 As one can see, all of {es} <<mapping-types, data types>> are mapped to the data type with the same
 name in {es-sql}, with the exception of **date** data type which is mapped to **datetime** in {es-sql}.
-This is to avoid confusion with the ANSI SQL **DATE** (date only) type, which is also supported by {es-sql}
-in queries (with the use of <<sql-functions-type-conversion-cast>>/<<sql-functions-type-conversion-convert>>),
-but doesn't correspond to an actual mapping in {es} (see the <<es-sql-only-types, `table`>> below).
+This is to avoid confusion with the ANSI SQL types **DATE** (date only) and **TIME** (time only), which is also
```
Comment (costin) [inline docs/reference/sql/language/data-types.asciidoc]: which is -> which are.

File: docs/reference/sql/limitations.asciidoc [hunk 2]
```diff
@@ -113,3 +113,28 @@ FROM (SELECT ...) WHERE [simple_condition]`, this is currently **un-supported**.
 Using `FIRST` and `LAST` in the `HAVING` clause is not supported. The same applies to
 <<sql-functions-aggs-min,`MIN`>> and <<sql-functions-aggs-max,`MAX`>> when their target column
 is of type <<keyword, `keyword`>> as they are internally translated to `FIRST` and `LAST`.
+
+[float]
+=== Using TIME data type in GROUP BY or <<sql-functions-grouping-histogram>>
+
+Using `TIME` data type as a grouping key is currently not supported. For example:
```
Comment (costin) [inline docs/reference/sql/limitations.asciidoc]: This should be a `NOTE:` or `WARNING:`

File: docs/reference/sql/limitations.asciidoc [hunk 3]
```diff
@@ -113,3 +113,28 @@ FROM (SELECT ...) WHERE [simple_condition]`, this is currently **un-supported**.
...
+
+[source, sql]
```
Comment (costin) [inline docs/reference/sql/limitations.asciidoc]: An alternative is to move these tests back to the docs.csv-spec and mark them as `-Ignore` which means they will not be ran.

**Review by costin — APPROVED** [review]
LGTM

File: docs/reference/sql/limitations.asciidoc (hunk 2 above)
Comment (matriv) [reply docs/reference/sql/limitations.asciidoc]: We don't use them in the limitations page. In the `grouping.asciidoc` there is and `IMPORTANT` paragraph mentioning the same limitation.

File: docs/reference/sql/functions/date-time.asciidoc [hunk 4]
```diff
@@ -139,6 +139,63 @@ is used for relative date filtering:
...
 --------------------------------------------------
 
+[[sql-functions-current-time]]
+==== `CURRENT_TIME/CURTIME`
+
+.Synopsis:
+[source, sql]
+--------------------------------------------------
+CURRENT_TIME
+CURRENT_TIME(precision <1>)
```
Comment (astefan) [inline docs/reference/sql/functions/date-time.asciidoc]: I thought we were using the `CURRENT_TIME([precision <1>])` (square brackets) for optional parameters, but looking at CURRENT_TIMESTAMP I see the same approach. Other functions (ROUND, TRUNCATE and few string functions) use the square brackets format.

File: docs/reference/sql/functions/date-time.asciidoc [hunk 5]
```diff
@@ -139,6 +139,63 @@ is used for relative date filtering:
...
+
+Typically, this function is used for relative date/time filtering:
+
+["source","sql",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{sql-specs}/docs/docs.csv-spec[filterCurrentTime]
+--------------------------------------------------
+
+[IMPORTANT]
+Currently, Using a _precision_ greater than 3 doesn't make any difference to the output of the
```
Comment (astefan) [inline docs/reference/sql/functions/date-time.asciidoc]: Please, lowercase `Using`.

**Review by astefan — APPROVED** [review]
LGTM. Left few comments regarding the documentation.

File: docs/reference/sql/functions/date-time.asciidoc [hunk 6]
```diff
@@ -139,6 +139,63 @@ is used for relative date filtering:
...
+include-tagged::{sql-specs}/docs/docs.csv-spec[currentTime]
+--------------------------------------------------
+
+["source","sql",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{sql-specs}/docs/docs.csv-spec[currentTimeFunction]
+--------------------------------------------------
+
+["source","sql",subs="attributes,callouts,macros"]
+--------------------------------------------------
```
[the remark this answers is not in the document]
Comment (matriv) [reply docs/reference/sql/functions/date-time.asciidoc]: Should be an example for `CURTIME`, fixing it.

File: docs/reference/sql/functions/date-time.asciidoc (hunk 4 above)
Comment (matriv) [reply docs/reference/sql/functions/date-time.asciidoc]: Adding the square brackets, will also fix it for CURRENT_TIMESTAMP with a separate PR.

**Pull request merged**